nyc/bills.py

```diff
@@ -161,133 +161,155 @@ def sponsorships(self, matter_id):
+
+        for topic in self.topics(matter_id) :
+            bill.add_subject(topic['MatterIndexName'].strip())
+
+        for relation in self.relations(matter_id):
             try:
-                for sponsorship in self.sponsorships(matter_id):
-                    bill.add_sponsorship(**sponsorship)
-            except KeyError:
-                version_errors.append(legistar_web)
-                continue
+                related_bill = self.endpoint('/matters/{0}', relation['MatterRelationMatterId'])
+            except scrapelib.HTTPError:
+                return None
+            else:
+                date = related_bill['MatterIntroDate']
+                related_bill_session = self.session(self.toTime(date))
+                identifier = related_bill['MatterFile']
+                bill.add_related_bill(identifier=identifier,
+                                      legislative_session=related_bill_session,
+                                      relation_type='companion')
 
-            for attachment in self.attachments(matter_id):
+        try:
+            text = self.text(matter_id)
+        except KeyError:
+            self.version_errors.append(legistar_web)
+            return None
 
-                if attachment['MatterAttachmentId'] == 103315:  # Duplicate
-                    continue
 
-                if attachment['MatterAttachmentName']:
-                    bill.add_document_link(attachment['MatterAttachmentName'],
-                                           attachment['MatterAttachmentHyperlink'],
-                                           media_type='application/pdf')
+        if text:
+            if text['MatterTextPlain']:
+                bill.extras['plain_text'] = text['MatterTextPlain'].replace(u'\u0000', '')
 
-            for topic in self.topics(matter_id) :
-                bill.add_subject(topic['MatterIndexName'].strip())
+            if text['MatterTextRtf']:
+                bill.extras['rtf_text'] = text['MatterTextRtf'].replace(u'\u0000', '')
 
-            for relation in self.relations(matter_id):
-                try:
-                    related_bill = self.endpoint('/matters/{0}', relation['MatterRelationMatterId'])
-                except scrapelib.HTTPError:
-                    continue
-                else:
-                    date = related_bill['MatterIntroDate']
-                    related_bill_session = self.session(self.toTime(date))
-                    identifier = related_bill['MatterFile']
-                    bill.add_related_bill(identifier=identifier,
-                                          legislative_session=related_bill_session,
-                                          relation_type='companion')
+        return bill
 
-            try:
-                text = self.text(matter_id)
-            except KeyError:
-                version_errors.append(legistar_web)
-                continue
+    def get_vote_event(self, bill, action):
```

Comment:
This doesn't seem like the quite right separation.

when you do 

```python
        act = bill.add_action(action['action_description'],
                              action['action_date'],
                              organization={'name': action['responsible_org']},
                              classification=action['classification'])
```

does the `act` object have a handle on the bill? if so, could you just pass in an `act`?

Reply:
We don't have a handle on the vote events there, but I agree the separation is a bit confusing. How about c9d9f37?